src/test/test-backwards-compatible.gd

```diff
@@ -0,0 +1,72 @@
+extends "res://addons/gut/test.gd"
+"""
+Tests backwards compatibility with older save formats.
+"""
+
+const TEMP_FILENAME := "test-ground-lucky.save"
+const TEMP_FILENAME_0517 := "test-ground-lucky-0517.save"
+
+func before_each() -> void:
+	PlayerSave.player_data_filename = "user://%s" % TEMP_FILENAME
+	PlayerSave.old_save.player_data_filename_0517 = "user://%s" % TEMP_FILENAME_0517
+	PlayerData.reset()
+	
+	var dir := Directory.new()
+	dir.copy("res://src/demo/turbofat-v0.0517.save", "user://%s" % TEMP_FILENAME_0517)
+	PlayerSave.load_player_data()
+
+
+func after_each() -> void:
+	PlayerData.history_size = 1000
+	var save_dir := Directory.new()
+	save_dir.open("user://")
+	save_dir.remove(TEMP_FILENAME)
+	save_dir.remove(TEMP_FILENAME_0517)
+
+
+func test_backwards_compatible_lost() -> void:
```

Comment:
Now that they're in their own class, these methods can have less repetitive names

Reply:
Changed to `test_lost_true`, `test_lost_false`, etc